weddineo-core/src/main/java/com/weddineo/configuration/filter/FirebaseFilter.java

```diff
@@ -0,0 +1,75 @@
+package com.weddineo.configuration.filter;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.stream.Collectors;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseAuthException;
+import com.google.firebase.auth.FirebaseToken;
+import com.weddineo.firebase.exception.WeddineoFirebaseRuntimeException;
+
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.util.StringUtils;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import lombok.Builder;
+
+@Builder
+public class FirebaseFilter extends OncePerRequestFilter {
+
+    private static String AUTH_HEADER = "Authorization";
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
+            throws ServletException, IOException {
+        String path = request.getRequestURI();
+        // TODO: Should be all public endpoints
+        if ("/auth/register".equals(path)) {
+            filterChain.doFilter(request, response);
+            return;
+        }
+
+        String authToken = request.getHeader(AUTH_HEADER);
+        if (!StringUtils.isEmpty(authToken)) {
+            Authentication auth;
+            try {
+                auth = getAuthentication(authToken);
+                SecurityContextHolder.getContext().setAuthentication(auth);
+            } catch (FirebaseAuthException e) {
+                throw new WeddineoFirebaseRuntimeException("Cannot connect to firebase auth", e);
+            }
+
+            logger.debug("Successfully Authenticated");
+        }
+        filterChain.doFilter(request, response);
+
+    }
+
+    private FirebaseToken verifyIdToken(String idToken) throws FirebaseAuthException {
+        if (StringUtils.isEmpty(idToken)) {
+            throw new IllegalArgumentException("Token is empty");
+        }
+        return FirebaseAuth.getInstance().verifyIdToken(idToken);
+    }
+
+    private Authentication getAuthentication(String idToken) throws FirebaseAuthException {
+        FirebaseToken token = verifyIdToken(idToken);
+        assert token != null;
```

Comment:
I wolud prefer `Optional.ofNullable().orElseThrow()` - with that construction you can throw excepetion that describes problem in better. Now AssertationError will be thrown here.

Reply:
done